wicketd/src/artifacts.rs

```diff
@@ -33,41 +42,64 @@ pub(crate) struct WicketdArtifactStore {
     log: slog::Logger,
     // NOTE: this is a `std::sync::Mutex` rather than a `tokio::sync::Mutex` because the critical
     // sections are extremely small.
-    artifacts: Arc<Mutex<DebugIgnore<HashMap<ArtifactId, BufList>>>>,
+    artifacts_with_plan: Arc<Mutex<ArtifactsWithPlan>>,
```

Comment:
This is going to conflict with #2381 (though the conflicts should be easy to address since they're both directionally doing the same thing)

Reply:
Ah, yeah, looks like we can drop in `UpdatePlan::new()` for `        // TODO: build out update planner here`, and add the plan to `TufRepoArtifacts`. Do you want to merge 2381 first?

Reply:
I think you should just land this first since it's ready to go, and I'll rebase on top of your change.